src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -178,6 +182,143 @@ export abstract class ToggleableAction extends Action {
 	}
 }
 
+export class NotebookViewsOptions implements IActionProvider {
+	private _options: Action[];
+	private views: NotebookViewsExtension;
+	private viewMode: ViewMode;
+	private readonly _optionsUpdated = new Emitter<boolean>();
+
+	constructor(
+		container: HTMLElement,
+		views: NotebookViewsExtension,
+		modelReady: Promise<INotebookModel>,
+		@INotebookService private _notebookService: INotebookService,
+		@IInstantiationService private instantiationService: IInstantiationService) {
+
+		if (modelReady) {
```

Comment:
The constructor should indicate that this is optional if you're expecting it to sometimes be undefined. 

And you can also just do `modelReady?.then(model => ...`) instead of the separate if check